vespalib/src/vespa/vespalib/regex/regex.h

```diff
@@ -44,7 +44,8 @@ class Regex {
     // TODO consider using type-safe parameter instead.
     enum Options {
         None       = 0,
-        IgnoreCase = 1
+        IgnoreCase = 1,
+        MultiLine  = 2
```

Comment:
According to the RE2 docs, `set_dot_nl(bool)` has the behavior "dot matches everything including new line". Maybe we should call this `DotMatchesNewline` instead to be explicit that this does not trigger any changes to how anchors work etc? `MultiLine` might imply some different semantics than what is currently in place.